### DISCUSSION TITLE
`<rh-table>`: sorting columns when there are merged rows

### DISCUSSION THREAD

**Discussion by bennypowers** [RedHat-UX/red-hat-design-system#1455] [Ideas]
When [`rowspan`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/td#rowspan) is in use, how should `<rh-sort-button>` behave? 

Take this table for example

<table>
  <thead>
    <th>ID</th>
    <th>Properties</th>
    <th>Values</th>
  </thead>
  <tbody>
   <tr>
     <td rowspan=2>1</td>
     <td>A</td>
     <td>first</td>
  </tr>
  <tr>
     <td>B</td>
     <td>second</td>
</tr>
   <tr>
     <td rowspan=3>2</td>
     <td>A</td>
     <td>first</td>
  </tr>
  <tr>
     <td>B</td>
     <td rowspan=2>n/a</td>
</tr>
  <tr>
     <td>C</td>
</tr>
</tbody>
</table>

If there's a sort button on the first column, how should it behave? on the second or third column?

**Comment by markcaron** [comment]
Interesting question! Wonder if @hellogreg or @nikkimk have any thoughts on this.

**Comment by hellogreg** [comment]
**I admit this is the first time I've even considered this scenario, so take the following as more spitballing than authoritative!**

In this case, the ID number is not a header itself and has no connection to the Properties and Value headers, so I don't think it's a matter of _which_ column it should sort by. I think it shouldn't do _anything_ to restack rows within those columns. If you had 10 more headers in the top row, which of them would you sort by? In other words, I think any sorts should only be triggered by the those headers above the respective column. (Now, as to whether those headers sort every row in the table top-to-bottom or just sort within multi-row groups when there are ID rowspans > 1, that's another discussion.)

Even if we did allow a sort on the ID numbers, we're putting a sort button in a `<td>` cell (which doesn't seem like a good idea to me) and not a `<th>` cell.

And if you made that ID value a `<th>`, we've now got a table with two-dimensional headers--which can have sorting, but probably not in this case, where we're considering restacking rows based on a button in a `<th>` with a scope of `row`. My gut says a sort button in a header with a scope of `row` should not re-stack rows, and a button within a header with a scope of `col` should not shift column order.

That's not to say we should _never_ allow sorting based on parent headers with rowspan/colspan on them. Maybe there's an argument that we can. But if we did, I think it should work something like this...

Assuming "Name" is sortable in the following table, I'd think it could sort the names by the values in the "Last" column, and then by the values in the "First" column. In other words, "Benny Affleck" and "Casey Affleck" will now be the first two entries. What it should _not_ do is rearrange those columns so that "First" comes before "Last" in LTR column order--at least that's my instinct.

<table>
  <thead>
    <tr>
      <th rowspan=2 scope="col">Employee ID</th>
      <th colspan=2 scope="col"><a href="#" role="button">Name</a></th>
    </tr>
    <tr>
      <th scope="col">Last</th>
      <th scope="col">First</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>001</td>
      <td>Powers</td>
      <td>Benny</td>
    </tr>
    <tr>
      <td>002</td>
      <td>Caron</td>
      <td>Mark</td>
    </tr>
    <tr>
      <td>003</td>
      <td>Affleck</td>
      <td>Casey</td>
    </tr>
        <tr>
      <td>004</td>
      <td>Affleck</td>
      <td>Benny</td>
    </tr>
  </tbody>
